metpy/plots/_util.py

```diff
@@ -31,6 +31,12 @@ def colored_line(x, y, c, **kwargs):
         The created :class:`matplotlib.collections.LineCollection` instance.
 
     """
+    # Mask out any NaN values
+    nan_mask = ~(np.isnan(x) | np.isnan(y) | np.isnan(c))
+    x = x[nan_mask]
```

Comment:
I'm wondering now if instead of just removing, we should be adding gaps, like in a standard MPL line plot. Thoughts?